staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5107,12 +5107,17 @@ type ReplicationControllerSpec struct {
 	// Defaults to 1.
 	// More info: https://kubernetes.io/docs/concepts/workloads/controllers/replicationcontroller#what-is-a-replicationcontroller
 	// +optional
+	// +k8s:optional
+	// +default=1
+	// +k8s:minimum=0
 	Replicas *int32 `json:"replicas,omitempty" protobuf:"varint,1,opt,name=replicas"`
 
 	// Minimum number of seconds for which a newly created pod should be ready
 	// without any of its container crashing, for it to be considered available.
 	// Defaults to 0 (pod will be considered available as soon as it is ready)
 	// +optional
+	// +default=0
```

Comment:
Now this is interesting.  @thockin in a theory of declaring a marker and that marker must be zero value, what does a default of the zero value mean?  For built-ins, this won't even impact serialization. Why isn't `+k8s:optional` more appropriate

Reply:
> Now this is interesting. @thockin in a theory of declaring a marker and that marker must be zero value, what does a default of the zero value mean? 

It just documents the behavior, I think?  I kinda like the explicitness of it-- without having to grok the field's type and the zero-valued behavior, this tells a user what to expect when they do not provide a value.

> For built-ins, this won't even impact serialization. Why isn't `+k8s:optional` more appropriate

@thockin, based on the [below comment](https://github.com/kubernetes/kubernetes/pull/130725#discussion_r1992473053), should we have `k8s:optional` here given that we have `optional`?